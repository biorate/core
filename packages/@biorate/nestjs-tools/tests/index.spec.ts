import { expect, use } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';
import {
  AuthBasic,
  Roles,
  UserRoles,
  AllExceptionsFilter,
  RoutesInterceptor,
  ProxyPrometheusMiddleware,
  RequestCountMiddleware,
  ResponseTimeMiddleware,
  AuthGuardProvider,
  RolesGuardProvider,
} from '../src';

use(jestSnapshotPlugin());

describe('nestjs-tools', () => {
  it('AuthBasic', () => expect(AuthBasic).toMatchSnapshot());

  it('Roles', () => expect(Roles).toMatchSnapshot());

  it('UserRoles', () => expect(UserRoles).toMatchSnapshot());

  it('AllExceptionsFilter', () => expect(AllExceptionsFilter).toMatchSnapshot());

  it('RoutesInterceptor', () => expect(RoutesInterceptor).toMatchSnapshot());

  it('ProxyPrometheusMiddleware', () =>
    expect(ProxyPrometheusMiddleware).toMatchSnapshot());

  it('RequestCountMiddleware', () => expect(RequestCountMiddleware).toMatchSnapshot());

  it('ResponseTimeMiddleware', () => expect(ResponseTimeMiddleware).toMatchSnapshot());

  it('AuthGuardProvider', () => expect(AuthGuardProvider).toMatchSnapshot());

  it('RolesGuardProvider', () => expect(RolesGuardProvider).toMatchSnapshot());
});
