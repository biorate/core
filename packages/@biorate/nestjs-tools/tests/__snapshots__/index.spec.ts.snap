// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nestjs-tools AllExceptionsFilter 1`] = `[Function]`;

exports[`nestjs-tools AuthBasic 1`] = `[Function]`;

exports[`nestjs-tools AuthGuardProvider 1`] = `[Function]`;

exports[`nestjs-tools GetLocalesDTO 1`] = `[Function]`;

exports[`nestjs-tools LocalesController 1`] = `[Function]`;

exports[`nestjs-tools MetricsController 1`] = `[Function]`;

exports[`nestjs-tools PostLocalesDTO 1`] = `[Function]`;

exports[`nestjs-tools ProbeController 1`] = `[Function]`;

exports[`nestjs-tools ProxyPrometheusMiddleware 1`] = `[Function]`;

exports[`nestjs-tools ROLES_KEY 1`] = `Symbol(RolesKey)`;

exports[`nestjs-tools RequestCountMiddleware 1`] = `[Function]`;

exports[`nestjs-tools ResponseTimeMiddleware 1`] = `[Function]`;

exports[`nestjs-tools Roles 1`] = `[Function]`;

exports[`nestjs-tools RolesGuardProvider 1`] = `[Function]`;

exports[`nestjs-tools RoutesInterceptor 1`] = `[Function]`;

exports[`nestjs-tools TracingInterceptor 1`] = `[Function]`;

exports[`nestjs-tools UnsupportedProtocolError 1`] = `[Function]`;

exports[`nestjs-tools UserRoles 1`] = `
Object {
  "1": "Basic",
  "2": "Admin",
  "Admin": 2,
  "Basic": 1,
}
`;
